wallet/chainntfns.go

```diff
@@ -174,6 +176,34 @@ func (w *Wallet) ChainSwitch(ctx context.Context, forest *SidechainForest, chain
 
 			// Add the block hash to the notification.
 			chainTipChanges.AttachedBlocks = append(chainTipChanges.AttachedBlocks, n.Hash)
+
+			if birthState != nil &&
+				((birthState.SetFromTime && n.Header.Timestamp.After(birthState.Time)) ||
+					(birthState.SetFromHeight && n.Header.Height == birthState.Height+1)) {
+				bh := n.Header.PrevBlock
```

Comment:
Wait maybe this doesnt work for the edge case where time was set before the genesis block.

Reply:
I tried it and it was fine, hmm.

Reply:
Maybe this starts from block 1?

Reply:
prints suggest it does.